### DISCUSSION TITLE
Determining Sample Size in Neural Receiver for OFDM MIMO

### DISCUSSION THREAD

**Discussion by yhuwhai** [NVlabs/sionna#240] [Q&A] [answered]
Hi everyone,

I am wondering, for the neural receiver example on sionna, my assumption is that the training sample size is given by (number of iterations x batch size).

For the evaluation sample size, how can this be determined? The batch size for both evaluation and training are the same (128) . There is a variable in the sim_ber function called max_mc_iter. Does changing this variable (max_mc_iter)  affect the evaluation sample size? 

I have attached a plot to show the effect of varying the max_mc_iter.

Thanks for your answer in advance. 

![image](https://github.com/NVlabs/sionna/assets/146411286/cb0cc03a-87a3-4425-a37e-762839d16d37)

**Comment by jhoydis** [comment, marked as answer]
For the evaluation, we adapt the number of iterations depending on multiple criteria to speed-up the Monte Carlo simulations.
If you set the parameters ``num_target_bit_errors`` and ``num_target_block_errors`` to ``None``,  the ``sim_ber`` function will evaluate for every SNR point exactly ``max_mc_iter`` batches. However, this is in most cases not needed and one can stop the simulation for a specific SNR point if either a certain number of bit or block errors has occured. If you set these values to low, your BER/BLER curves will be too noisy. Setting them too large will not improve accuracy much. The art is to find the right value.

**Reply by yhuwhai** [reply]
Yes. Thanks. Setting large values will guarantee smother curves.

On the issue of training sample size, is my thinking on track ?

**Reply by jhoydis** [reply]
Yes.
